jupyterlab_ucaip/src/components/datasets/dataset_component.tsx

```diff
@@ -96,6 +132,40 @@ export class DatasetComponent extends React.Component<Props, State> {
     }
   }
 
+  private async trainModel() {
+    try {
+      this.setState({ isLoading: true });
+      const displayName = this.props.dataset.displayName;
+      const transformations = [];
+      for (let i = 0; i < this.state.columns.length; i++) {
+        if (this.state.columns[i].fieldName !== this.state.targetColumn) {
+          const column = { columnName: this.state.columns[i].fieldName };
+          transformations.push({ auto: column });
+        }
+      }
+      await DatasetService.trainModel(
+        displayName + '_training',
+        this.getDatasetId(),
+        displayName + '_model',
+        this.state.targetColumn,
+        this.state.predictionType,
+        this.state.objective,
+        parseInt(this.state.budget),
+        transformations
+      );
+      this.setState({ trainingResponse: true });
+    } catch (err) {
+      console.warn('Error creating training pipeline', err);
+    } finally {
+      this.setState({ isLoading: false });
+    }
+  }
+
+  private getDatasetId(): string {
+    const datasetId = this.props.dataset.id.split('/');
```

Comment:
Is this variable appropriately named? Is the whole thing the dataset ID or the first part? Would a name like ```datasetIdParts``` be more appropriate?